File: go/leap/leap.go
// Package leap provides testing if a year is a leap your or not
package leap

// IsLeapYear reports if a year is a leap year or not
func IsLeapYear(year int) bool {
	if year%400 == 0 {
		return true
	}
	if year%100 == 0 {
		return false
	}
	return year%4 == 0
}

File: fsharp/leap/Leap.fs
module Leap

let leapYear year =
    year % 400 = 0 || (year % 4 = 0 && year % 100 <> 0)
    // function
    // | year when year % 400 = 0 -> true
    // | year when year % 100 = 0 -> false
    // | year when year % 4 = 0 -> true
    // | _ -> false